src/core/qgsclipper.h

```diff
@@ -99,19 +104,36 @@ class CORE_EXPORT QgsClipper
 
     SIP_END
 
+    /**
+     * Trims the given polygon to a rectangular box. Modify the given polygon.
+     *
+     * \param pts polygon as 2D coordinates
+     * \param clipRect clipping rectangle
+     */
     static void trimPolygon( QPolygonF &pts, const QgsRectangle &clipRect );
 
     /**
-     * Takes a linestring and clips it to clipExtent
+     * Trims the given polygon to a pseudo 3D box. Modify the given polygon.
+     *
+     * \param pts polygon as 2D or 3D coordinates
+     * \param clipRect clipping 3D box
+     */
+    static void trimPolygon( QgsLineString &pts, const QgsBox3d &clipRect );
```

Comment:
```suggestion
    static void trimPolygon( QgsLineString &polygon, const QgsBox3d &clipRect );
```

May as well improve the argument naming for this new method :+1: